File: backend/app/schemas/incidencia.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from app.models.incidencia import EstadoIncidencia, PrioridadIncidencia

class IncidenciaBase(BaseModel):
    titulo: str = Field(..., max_length=200, description="Título de la incidencia")
    descripcion: Optional[str] = Field(None, description="Descripción detallada")
    prioridad: PrioridadIncidencia = Field(PrioridadIncidencia.MEDIA, description="Prioridad: baja, media, alta, urgente")

class IncidenciaCreate(IncidenciaBase):
    inmueble_id: int = Field(..., description="ID del inmueble asociado")

class IncidenciaUpdate(BaseModel):
    titulo: Optional[str] = Field(None, max_length=200)
    descripcion: Optional[str] = None
    estado: Optional[EstadoIncidencia] = Field(None, description="Estado: abierta, asignada, en_progreso, resuelta, cerrada")
    prioridad: Optional[PrioridadIncidencia] = None
    proveedor_id: Optional[int] = None
    version: Optional[int] = None
    comentario_cambio: Optional[str] = Field(None, description="Comentario para el historial de cambios")

class HistorialIncidenciaResponse(BaseModel):
    id: int
    estado_anterior: Optional[str]
    estado_nuevo: str
    comentario: Optional[str]
    fecha: datetime
    usuario_nombre: Optional[str] = None
    
    class Config:
        from_attributes = True

class InmuebleSimple(BaseModel):
    id: int
    referencia: str
    direccion: Optional[str] = None
    
    class Config:
        from_attributes = True

class IncidenciaResponse(IncidenciaBase):
    id: int
    inmueble_id: int
    creador_usuario_id: int
    proveedor_id: Optional[int]
    estado: EstadoIncidencia
    fecha_alta: datetime
    fecha_cierre: Optional[datetime]
    version: int
    creado_en: datetime
    inmueble: Optional[InmuebleSimple] = None
    historial: List[HistorialIncidenciaResponse] = []
    
    class Config:
        from_attributes = True

File: mobile/app/src/main/java/com/tomoko/fyntra/data/models/Incidencia.kt
package com.tomoko.fyntra.data.models

data class Incidencia(
    val id: Int,
    val titulo: String,
    val descripcion: String? = null,
    val prioridad: String,
    val estado: String,
    val inmueble_id: Int,
    val creador_usuario_id: Int,
    val proveedor_id: Int? = null,
    val fecha_alta: String,
    val fecha_cierre: String? = null,
    val version: Int,
    val inmueble: InmuebleSimple? = null,
    val proveedor: ProveedorSimple? = null,
    val historial: List<HistorialIncidencia>? = null,
    val actuaciones_count: Int = 0,
    val documentos_count: Int = 0,
    val mensajes_count: Int = 0
)

data class ProveedorSimple(
    val id: Int,
    val nombre: String,
    val email: String? = null,
    val telefono: String? = null,
    val especialidad: String? = null
)

data class InmuebleSimple(
    val id: Int,
    val referencia: String,
    val direccion: String
)

data class HistorialIncidencia(
    val id: Int,
    val estado_anterior: String? = null,
    val estado_nuevo: String,
    val comentario: String? = null,
    val fecha: String,
    val usuario_nombre: String? = null
)

data class IncidenciaCreate(
    val titulo: String,
    val descripcion: String,
    val prioridad: String,
    val inmueble_id: Int
)

data class IncidenciaUpdate(
    val titulo: String? = null,
    val descripcion: String? = null,
    val prioridad: String? = null,
    val estado: String? = null,
    val proveedor_id: Int? = null,
    val comentario_cambio: String? = null,
    val version: Int
)

